cse/indexing/MainPostingListIndex.py

```diff
@@ -0,0 +1,103 @@
+import os
+from os import remove
+from tempfile import mkstemp
+from shutil import move
+
+
+"""
+Inverted Index (main - on disk)
+Structure: Line Number in File -> Posting List
+"""
+class MainPostingListIndex(object):
+
+    __postingLists = None
+    __postingListsFilename = ""
+
+
+    def __init__(self, filename):
+        self.__postingListsFilename = filename
+        self.__open()
+
+
+    def __open(self):
+        if not os.path.exists(os.path.dirname(self.__postingListsFilename)):
+            try:
+                os.makedirs(os.path.dirname(self.__postingListsFilename))
+            except OSError as exc: # Guard against race condition
+                if exc.errno != errno.EEXIST:
+                    raise
+
+        if not os.path.exists(self.__postingListsFilename):
+            print("postinglist file not available...creating file")
+            os.mknod(self.__postingListsFilename)
+        self.__postingLists = open(self.__postingListsFilename, 'r', newline='', encoding="utf-8")
+
+
+    def __decodePlLine(self, line):
+        return list(line.replace("\n", "").split(","))
+
+
+    def __encodePlLine(self, postingList):
+        return ",".join(postingList) + "\n"
+
+
+    def close(self):
+        self.save()
+
+
+    def save(self):
+        self.__postingLists.close()
+
+
+    def retrieve(self, pointer):
+        self.__postingLists.seek(0)
+        for i, plLine in enumerate(self.__postingLists):
+            if i == pointer:
+                return self.__decodePlLine(plLine)
+        return None
+
+
+    def estimatedSize(self):
+        return self.__sizeof__()
+
+
+    def mergeInDeltaIndex(self, dIndex):
+        self.__postingLists.seek(0)
+        _, tempFilePath = mkstemp()
+        visited = set()
+        with open(tempFilePath, 'w', newline='', encoding="utf-8") as tempFile:
```

Comment:
In L66 you already got an filehandle (`_` - the name could be more significant).  Why open the file again?

Reply:
this is a copy paste issue
Should be changed!